tests/unit/commands/local/lib/test_sam_api_provider.py

```diff
@@ -965,6 +953,198 @@ def test_binary_media_types_with_rest_api_id_reference(self):
         assertCountEqual(self, expected_apis, provider.apis)
 
 
+class TestSamApiProviderwithApiGatewayRestApi(TestCase):
+
+    def setUp(self):
+        self.binary_types = ["image/png", "image/jpg"]
+        self.input_apis = [
+            Api(path="/path1", method="GET", function_name="SamFunc1", cors=None),
+            Api(path="/path1", method="POST", function_name="SamFunc1", cors=None),
+
+            Api(path="/path2", method="PUT", function_name="SamFunc1", cors=None),
+            Api(path="/path2", method="GET", function_name="SamFunc1", cors=None),
+
+            Api(path="/path3", method="DELETE", function_name="SamFunc1", cors=None)
+        ]
+
+    def test_with_no_apis(self):
+        template = {
+            "Resources": {
+
+                "Api1": {
+                    "Type": "AWS::ApiGateway::RestApi",
+                    "Properties": {
+                        "StageName": "Prod"
+                    },
+
+                }
+            }
+        }
+
+        provider = SamApiProvider(template)
+
+        self.assertEquals(len(provider.apis), 0)
+        self.assertEquals(provider.apis, [])
```

Comment:
You don't need to assert the length here. Asserting that it is an empty list is sufficient.